CMake/FindArmadillo.cmake

```diff
@@ -52,318 +52,102 @@ if(ARMADILLO_INCLUDE_DIR)
     VERSION_VAR ARMADILLO_VERSION_STRING)
-  # version_var fails with cmake < 2.8.4.
-else ()
-  # Checks 'REQUIRED', 'QUIET' and versions.
-  include(FindPackageHandleStandardArgs)
+else(_ARMA_USE_WRAPPER)
+  # don't link to armadillo in this case
+  set(ARMADILLO_LIBRARY "")
+  if(_ARMA_USE_LAPACK)
+    find_package(LAPACK REQUIRED)
+    set(_ARMA_SUPPORT_LIBRARIES "${_ARMA_SUPPORT_LIBRARIES}" "${LAPACK_LIBRARIES}")
+  endif(_ARMA_USE_LAPACK)
+  if(_ARMA_USE_BLAS)
+    find_package(BLAS REQUIRED)
+    set(_ARMA_SUPPORT_LIBRARIES "${_ARMA_SUPPORT_LIBRARIES}" "${BLAS_LIBRARIES}")
+  endif(_ARMA_USE_BLAS)
+  if(_ARMA_USE_ARPACK)
+    find_package(ARPACK REQUIRED)
+    set(_ARMA_SUPPORT_LIBRARIES "${_ARMA_SUPPORT_LIBRARIES}" "${ARPACK_LIBRARIES}")
+  endif(_ARMA_USE_ARPACK)
+  if(_ARMA_USE_HDF5)
+    find_package(HDF5 QUIET)
+    if(NOT HDF5_FOUND)
+      # On Debian systems, the HDF5 package has been split into multiple
+      # packages so that it is co-installable.  But this may mean that the
+      # include files are hidden somewhere very odd that FindHDF5.cmake  will
+      # not find.  Thus, we'll also quickly check pkgconfig to see if there is
+      # information on what to use there.
+      message(WARNING "HDF5 required but not found; using PkgConfig")
+      find_package(PkgConfig)
+      if (PKG_CONFIG_FOUND)
+        pkg_check_modules(HDF5 REQUIRED hdf5)
+        link_directories("${HDF5_LIBRARY_DIRS}")
+      else()
+        message(FATAL_ERROR "PkgConfig (Used to help find HDF5) was not found")
+      endif()
+    endif()
+    set(_ARMA_SUPPORT_INCLUDE_DIRS "${HDF5_INCLUDE_DIRS}")
+    set(_ARMA_SUPPORT_LIBRARIES "${_ARMA_SUPPORT_LIBRARIES}" "${HDF5_LIBRARIES}")
+  endif(_ARMA_USE_HDF5)
+  set(ARMADILLO_FOUND true)
   find_package_handle_standard_args(Armadillo
     REQUIRED_VARS ARMADILLO_INCLUDE_DIR
     VERSION_VAR ARMADILLO_VERSION_STRING)
-endif ()
+endif(_ARMA_USE_WRAPPER)
 
 if (ARMADILLO_FOUND)
-  # Also include support include directories.
-  set(ARMADILLO_INCLUDE_DIRS ${ARMADILLO_INCLUDE_DIR} ${SUPPORT_INCLUDE_DIRS})
-  # Also include support libraries to link against.
-  if (ARMA_NEED_LIBRARY)
-    set(ARMADILLO_LIBRARIES ${ARMADILLO_LIBRARY} ${SUPPORT_LIBRARIES})
-  else ()
-    set(ARMADILLO_LIBRARIES ${SUPPORT_LIBRARIES})
-  endif ()
-  message(STATUS "Armadillo libraries: ${ARMADILLO_LIBRARIES}")
+  set(ARMADILLO_INCLUDE_DIRS ${ARMADILLO_INCLUDE_DIR})
+  set(ARMADILLO_INCLUDE_DIRS ${ARMADILLO_INCLUDE_DIRS} ${_ARMA_SUPPORT_INCLUDE_DIRS})
+  set(ARMADILLO_LIBRARIES ${ARMADILLO_LIBRARY} ${_ARMA_SUPPORT_LIBRARIES})
 endif ()
 
+# Clean up internal variables
+unset(_ARMA_SUPPORT_LIBRARIES)
+unset(_ARMA_USE_WRAPPER)
+unset(_ARMA_USE_LAPACK)
+unset(_ARMA_USE_BLAS)
+unset(_ARMA_USE_ARPACK)
+unset(_ARMA_USE_HDF5)
+unset(_ARMA_CONFIG_CONTENTS)
+unset(_ARMA_HEADER_CONTENTS)
+unset(_ARMA_SUPPORT_INCLUDE_DIRS)
 
 # Hide internal variables
 mark_as_advanced(
+  _ARMA_SUPPORT_LIBRARIES
```

Comment:
If we just unset `_ARMA_SUPPORT_LIBRARIES`, do we need to call `mark_as_advanced()` with it?  Anyway I don't think it makes a huge difference either way.

Reply:
No; I forgot to remove it when I moved it to unset.